### PR TITLE
Adding a wrapper for keccak256 to allow hashing  `string` and `bigint`

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
Fixes https://github.com/ChainSafe/web3.js/issues/5287 by adding a wrapper for keccak256 to allow hashing  `string` and `bigint`. 
This MR also includes test cases.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-utils/src/hash.ts [hunk 1]
```diff
@@ -57,7 +57,25 @@ export const sha3Raw = (data: Bytes): string => {
 	return hash;
 };
 
-export { keccak256 };
+/**
+ * A wrapper for ethereum-cryptography/keccak256 to allow hashing a `string` and a `bigint` in addition to `UInt8Array`
+ */
+export const keccak256Wrapper = (
+	data: Uint8Array | ReadonlyArray<number> | string | bigint,
```
Comment (nazarhussain) [inline packages/web3-utils/src/hash.ts]: Prefer to use the union types we have. 


```suggestion
	data: Bytes | Numbers | string,
```

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Review by jdevcs — APPROVED** [review]
apart from Nazar's comments LGTM.

File: packages/web3-utils/src/hash.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-utils/src/hash.ts]: Sorry, this is incompatible with keccak256. In web3.js v1.x, It throws an exception if we pass a number that is provided from type `Numbers` or if we pass `ArrayBuffer` that is provided from type `Bytes`. 
Additionally, the `readonly numbers[]` is supported as I am using `Buffer.from(...)` and so I included it.

**Review by luu-alex — APPROVED** [review]

File: packages/web3-utils/src/hash.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-utils/src/hash.ts]: As per our discussion, it is nice to support Bytes and Numbers, similar to other utility functions. 
So, I modified the logic to include more types as desired. Kindly re-check now.

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**